coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/consumer/AbstractStreamConsumer.java

```diff
@@ -0,0 +1,185 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.module.redisstream.consumer;
+
+import java.text.MessageFormat;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Inject;
+
+import org.jboss.weld.context.bound.BoundRequestContext;
+
+import hu.icellmobilsoft.coffee.dto.exception.BaseException;
+import hu.icellmobilsoft.coffee.module.redis.annotation.RedisConnection;
+import hu.icellmobilsoft.coffee.module.redisstream.service.RedisStreamService;
+import hu.icellmobilsoft.coffee.se.logging.Logger;
+import hu.icellmobilsoft.coffee.tool.utils.string.RandomUtil;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.StreamEntry;
+
+/**
+ * Abstract stream consumer class
+ * 
+ * @author imre.scheffer
+ * @since 1.3.0
+ */
+public abstract class AbstractStreamConsumer implements IRedisStreamConsumer {
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private RedisStreamService redisStreamService;
+
+    private String consumerIdentifier;
+
+    private String redisConfigKey;
+
+    private boolean endLoop;
+
+    @Inject
+    private BoundRequestContext boundRequestContext;
+
+    @Override
+    public void init(String redisConfigKey, String group) {
+        this.redisConfigKey = redisConfigKey;
+        redisStreamService.setGroup(group);
+    }
+
+    /**
+     * Vegtelen ciklus inditasa, ami a streamet olvassa
+     */
+    public void startLoop() {
+        consumerIdentifier = RandomUtil.generateId();
+        endLoop = false;
+        boolean firstRun = true;
+        while (!endLoop) {
+            Optional<StreamEntry> streamEntry = Optional.empty();
+            Instance<Jedis> jedisInstance = CDI.current().select(Jedis.class, new RedisConnection.Literal(redisConfigKey));
+            Jedis jedis = null;
+            try {
+                jedis = jedisInstance.get();
+                redisStreamService.setJedis(jedis);
+
+                if (firstRun) {
+                    // lehethogy a csoport nem letezik
+                    redisStreamService.handleGroup();
```

Comment:
BIztos elég ha csak firstRun-nál megy ez az ellenőrzés? Mi történik, ha pl újraindítják a redis-t; ha olyankor eltűnik a group, akkor gondolom minden consume Exception-re fog futni, amíg nem indul egy uj service, aki létrehozza.

Reply:
Ugylatom a default redis docker nem persistent modban fut fel, es akkor is a stream dolgok mind megmaradnak benne. Szoval egyenlore ez jonak tunik.
Az hogy minden ciklusban legyen ellenorizve, nagy rahatassal lenne a sebeseggre, amit mindenkepp kerulni kell. Ha nem lessz jo igy akkor majd valami hibakezeles uton modositjuk

Reply:
Minden ciklusban ellenőrizni nem jó ut ebben egyetértek, inkább arra gondoltam, hogy catch blokkban lehetne megpróbálni kezelni, ha a group miatt száll el. Viszont, ha a redis megtartja a dolgokat, akkor tényleg jó lehet így.